spinalcordtoolbox/scripts/sct_maths.py

```diff
@@ -408,152 +419,169 @@ def main(argv: Sequence[str]):
 
     # Actually do the computations
     im = Image(arguments.i)
-    data = im.data
-    for arg_name, arg_value in arguments.todo:
-        assert data is not None
-
-        if arg_name == "add":
-            if data.ndim == 4 and not arg_value:
-                # special case to sum a 4D volume across the t axis
-                data = np.sum(data, axis=3)
-            else:
+
+    # If volumewise is specified, treat 4D image as a set of 3D volumes
+    if arguments.volumewise:
+        data_list = [np.squeeze(arr, axis=3) for arr in np.array_split(im.data, im.data.shape[3], 3)]
+        dim = list(im.dim)
```

Comment:
This actually clobbers the `dim` value computed from `arguments.dim` above the loop. Thankfully, the current version of the operations loop uses `data.shape` and `data.ndim` directly, so there's no need to track the array shape separately.